RiotSwiftUI/Modules/Authentication/Common/Service/MatrixSDK/AuthenticationState.swift

```diff
@@ -66,5 +66,24 @@ struct AuthenticationState {
                                              showRegistrationForm: registrationFlow != nil,
                                              ssoIdentityProviders: preferredLoginMode.ssoIdentityProviders ?? [])
         }
+
+        /// Needs authentication fallback for login or registration
+        var needsFallback: Bool {
```

Comment:
Do you think we should split this up into `needsLoginFallback` and `needsRegistrationFallback`? It would be sad to show it for both if it was only needed for one.